doc/tutorial/laminar.md

```diff
@@ -0,0 +1,675 @@
+
+Additionally, start Vite's development server if it wasn't already running:
+
+{% highlight shell %}
+$ npm run dev
+[...]
+{% endhighlight %}
+
+We can now change the contents of `src/main/scala/livechart/LiveChart.scala` to use Laminar instead of vanilla DOM APIs.
+
+At the top, we use the following import:
+
+{% highlight scala %}
+import com.raquo.laminar.api.L.{*, given}
+{% endhighlight %}
+
+That import brings all the Laminar features we want into scope.
+
+We replace the contents of the main `def LiveChart` method with a call to Laminar's `renderOnDomContentLoaded` method.
+This method bootstraps Laminar by installing a Laminar `Element` in an existing DOM element:
+
+{% highlight scala %}
+@main
+def LiveChart(): Unit =
+  renderOnDomContentLoaded(
+    dom.document.getElementById("app"),
+    Main.appElement()
+  )
+{% endhighlight %}
+
+and we then define `Main.appElement` as follows:
+
+{% highlight scala %}
+object Main:
+  def appElement(): Element =
+    div(
+      a(href := "https://vitejs.dev", target := "_blank",
+        img(src := "/vite.svg", className := "logo", alt := "Vite logo"),
+      ),
+      a(href := "https://developer.mozilla.org/en-US/docs/Web/JavaScript", target := "_blank",
+        img(src := javascriptLogo, className := "logo vanilla", alt := "JavaScript logo"),
+      ),
+      h1("Hello Laminar!"),
+      div(className := "card",
+        button(tpe := "button"),
+      ),
+      p(className := "read-the-docs",
+        "Click on the Vite logo to learn more",
+      ),
+    )
+  end appElement
+end Main
+{% endhighlight %}
+
+Instead of using HTML tags in a big string, we use Laminar functions corresponding to every kind of DOM element.
+In the parameters, we can specify both attributes, using the `:=` operator, and child elements.
+
+Every reference is type-checked.
+If we misspell `button` as `buton`, or if we try to put an integer in `target := 0`, the compiler will flag it as an error.
+
+### Code of the button
+
+Now, we have not yet replicated the behavior of the button.
+Previously the code using the DOM API was the following:
+
+{% highlight scala %}
+def setupCounter(element: dom.Element): Unit =
+  var counter = 0
+
+  def setCounter(count: Int): Unit =
+    counter = count
+    element.innerHTML = s"count is $counter"
+
+  element.addEventListener("click", e => setCounter(counter + 1))
+  setCounter(0)
+end setupCounter
+{% endhighlight %}
+
+In Laminar, we do not use event listeners to directly mutate the attributes of DOM elements.
+Instead, we use `Var`s and `Signal`s to model *time-varying* values.
+
+Let us isolate the button definition in a separate method, and illustrate how to write its behavior using Laminar:
+
+{% highlight scala %}
+  def counterButton(): Element =
+    val counter = Var(0)
+    button(
+      tpe := "button",
+      "count is ",
+      child.text <-- counter,
```

Comment:
I find this confusing: Is the append implicit / does it just add two text nodes?

Reply:
It creates two text nodes.

I can also use `child.text <-- counter.map(c => s"count is $c)`, but that requires to already explain `map` at that point, which doesn't fit the flow of the tutorial.